### PR TITLE
Add two Questions in Computer Science

### PR DESCRIPTION
**Pull request by khairalanam**
## Summary of changes

This PR adds two questions in Computer Science Quiz:

* Question on Hash table time complexity
* Question on N Queens Problem

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

### CODE REVIEW

File: src/data/general-cs-quiz.ts [hunk 1]
```diff
@@ -1792,17 +1815,6 @@ const generalCSQuiz = [
...
-  {
-    Question:
-      "Which data structure allows for efficient retrieval of data based on a key?",
-    Answer: "Hash Table",
-    Distractor1: "Binary Tree",
-    Distractor2: "Linked List",
-    Distractor3: "Queue",
-    Explanation:
-      "A hash table allows for efficient data retrieval based on a key through the use of a hash function.",
-    Link: "https://en.wikipedia.org/wiki/Hash_table"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1805]: why was this deleted?

File: src/data/general-cs-quiz.ts [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+    Distractor1: "O(logn)",
+    Distractor2: "O(n)",
+    Distractor3: "O(n^2)",
+    Explanation:
+      "The average time complexity of an element insertion in a hash table is O(1). Unlike arrays, hash tables use a hashing function to generate a unique key or index for each element, ensuring constant time for element insertion.",
+    Link: "https://www.freecodecamp.org/news/javascript-hash-table-associative-array-hashing-in-js/"
+  },
+  {
+    Question:
+      "Which of the following methods can be used to solve N-Queens Problem?",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:15]:
```suggestion
      "Which of the following methods can be used to solve the N-Queens problem?",
```

File: src/data/general-cs-quiz.ts [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+    Answer: "Backtracking",
+    Distractor1: "Greedy Paradigm",
+    Distractor2: "Divide and Conquer",
+    Distractor3: "Dynamic Programming",
+    Explanation:
+      "We use backtracking to solve N-Queens problem based on the condition that no two queens can eb present in the same row, column or diagonal.",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:21]:
```suggestion
      "Backtracking is used to solve the N-Queens problem based on the condition that no two queens can be present in the same row, column or diagonal.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

File: src/data/general-cs-quiz.ts (hunk 1 above)
Comment (khairalanam) [reply src/data/general-cs-quiz.ts:1805]: Sorry, my bad. While I was resolving conflicts (frankly speaking I have no idea how I did it), one of the questions got deleted. I will add the question.

Comment (khairalanam) [reply src/data/general-cs-quiz.ts]: I have added the question.

**Review by jdwilkin4 — APPROVED** [review]
Congrats on another contribution to our repository. 🎉
We look forward to future contributions. 👍

**Comment by khairalanam** [comment]
Thank you :)
